### PR TITLE
Makefile: add unconvert to inspect unnecessary type convert

### LINKED ISSUE #16264: Enhance make dev command to support unnecessary type conversion check
## Development Task

Related PR:  https://github.com/pingcap/tidb/pull/16262#issuecomment-611610499

We should add  static check to avoid unnecessary type conversion.

Maybe when we run make dev, we can inspection unnecessary type conversion.

### PR DESCRIPTION
**Pull request by Rustin-Liu**
Signed-off-by: Rustin-Liu <rustin.liu@gmail.com>

### What problem does this PR solve?

Issue Number: close #16264
Related PR:  https://github.com/pingcap/tidb/pull/16262#issuecomment-611610499

Problem Summary:
Codebase have many unnecessary type convert.
### What is changed and how it works?

What's Changed:

Add unconvert tool to inspect unnecessary type convert.

How it Works:
make check

### Related changes

None

### Check List 

Tests 

- Unit test
- Integration test

Side effects
None

### Release note 
None

### CODE REVIEW

**Comment by Rustin-Liu** [comment]
@sre-bot /run-all-tests

**Comment by Rustin-Liu** [comment]
@sre-bot /run-all-tests

**Comment by Rustin-Liu** [comment]
@sre-bot /run-all-tests

**Comment by Rustin-Liu** [comment]
@sre-bot /run-all-tests

**Comment by Rustin-Liu** [comment]
@sre-bot /run-all-tests

**Comment by Rustin-Liu** [comment]
@sre-bot /run-all-tests

**Comment by Rustin-Liu** [comment]
@zz-jason  PTAL~  
it seems check dev failed because go.sum changed.

**Comment by Rustin-Liu** [comment]
@sre-bot /run-all-tests

**Comment by Rustin-Liu** [comment]
@sre-bot /run-all-tests

**Comment by Rustin-Liu** [comment]
@sre-bot /run-all-tests

**Comment by Rustin-Liu** [comment]
@zz-jason  PTAL~  
it seems we have some unstable tests.

File: Makefile [hunk 1]
```diff
@@ -82,7 +83,7 @@ build:
 # Install the check tools.
 check-setup:tools/bin/revive tools/bin/goword tools/bin/gometalinter tools/bin/gosec
 
-check: fmt errcheck lint tidy testSuite check-static vet staticcheck
+check: fmt errcheck lint tidy testSuite check-static vet staticcheck unconvert
```
Comment (zz-jason) [inline Makefile:86]:
```suggestion
check: fmt errcheck unconvert lint tidy testSuite check-static vet staticcheck
```

Comment (Rustin-Liu) [reply Makefile]: Fixed.

File: Makefile [hunk 2]
```diff
@@ -128,6 +129,10 @@ staticcheck:
 	$(GO) get honnef.co/go/tools/cmd/staticcheck
 	$(STATICCHECK) ./...
 
+unconvert:
+	$(GO) get github.com/mdempsky/unconvert
```
Comment (tiancaiamao) [inline Makefile:133]: Can we move this to `tools/bin`? 
The `go get` command would pollute the `go.mod` file if it's put here.
See how other tools are managed for example.

**Comment by Rustin-Liu** [comment]
@sre-bot /run-all-tests

File: Makefile (hunk 2 above)
Comment (Rustin-Liu) [reply Makefile]: Fixed.

**Comment by Rustin-Liu** [comment]
@tiancaiamao  @zz-jason  PTAL~

**Comment by Rustin-Liu** [comment]
@tiancaiamao  Ping~

**Comment by Rustin-Liu** [comment]
@sre-bot /run-all-tests

**Comment by Rustin-Liu** [comment]
@sre-bot /run-all-tests

**Comment by tiancaiamao** [comment]
LGTM

**Pull request merged**